src/kokkos/plugin-SiPixelClusterizer/kokkos/gpuClustering.h

```diff
@@ -32,277 +30,253 @@ namespace KOKKOS_NAMESPACE {
         --j;
       if (j < 0 or id[j] != id[index]) {
         // boundary... replacing atomicInc with explicit logic
-        auto loc = Kokkos::atomic_fetch_add(&moduleStart[0], 1);
-        assert(moduleStart[0] < ::gpuClustering::MaxNumModules);
-        moduleStart[loc + 1] = index;
+        auto loc = Kokkos::atomic_fetch_add(&moduleStart(0),1);
+        assert(moduleStart(0) < ::gpuClustering::MaxNumModules);
+        moduleStart(loc + 1) = index;
       }
     }
-#ifdef TODO
-    __global__
-        //  __launch_bounds__(256,4)
-        void
-        findClus(uint16_t const* __restrict__ id,           // module id of each pixel
-                 uint16_t const* __restrict__ x,            // local coordinates of each pixel
-                 uint16_t const* __restrict__ y,            //
-                 uint32_t const* __restrict__ moduleStart,  // index of the first pixel of each module
-                 uint32_t* __restrict__ nClustersInModule,  // output: number of clusters found in each module
-                 uint32_t* __restrict__ moduleId,           // output: module id of each module
-                 int32_t* __restrict__ clusterId,           // output: cluster id of each pixel
-                 int numElements) {
-      if (blockIdx.x >= moduleStart[0])
-        return;
-
-      auto firstPixel = moduleStart[1 + blockIdx.x];
-      auto thisModuleId = id[firstPixel];
-      assert(thisModuleId < MaxNumModules);
-
-#ifdef GPU_DEBUG
-      if (thisModuleId % 100 == 1)
-        if (threadIdx.x == 0)
-          printf("start clusterizer for module %d in block %d\n", thisModuleId, blockIdx.x);
-#endif
 
-      auto first = firstPixel + threadIdx.x;
-
-      // find the index of the first pixel not belonging to this module (or invalid)
-      __shared__ int msize;
-      msize = numElements;
-      __syncthreads();
-
-      // skip threads not associated to an existing pixel
-      for (int i = first; i < numElements; i += blockDim.x) {
-        if (id[i] == InvId)  // skip invalid pixels
-          continue;
-        if (id[i] != thisModuleId) {  // find the first pixel in a different module
-          atomicMin(&msize, i);
-          break;
-        }
-      }
+    //  __launch_bounds__(256,4)
+    template <typename ExecSpace>
+    void findClus(Kokkos::View<uint16_t*, ExecSpace> id,           // module id of each pixel
```

Comment:
Here too the function should be moved out of the outermost `KOKKOS_NAMESPACE` namespace. On the other hand, then the `countModules()` above should be moved and templated as well, so maybe it is better to do that cleanup (also for `clusterChargeCut()` in a separate PR.